kore/src/Kore/Repl/State.hs

```diff
@@ -0,0 +1,543 @@
+        . Set.fromList
+        . Map.keys
+        <$> Lens.use lensAliases
+
+    falseToError :: AliasError -> Bool -> m ()
+    falseToError e b =
+        if b then pure () else Monad.Error.throwError e
+
+
+findAlias
+    :: MonadState (ReplState claim) m
+    => String
+    -> m (Maybe AliasDefinition)
+findAlias name = Map.lookup name <$> Lens.use lensAliases
+
+substituteAlias
+    :: AliasDefinition
+    -> ReplAlias
+    -> String
+substituteAlias
+    AliasDefinition { arguments, command }
+    ReplAlias { arguments = actualArguments } =
+    unwords
+      . fmap replaceArguments
+      . words
+      $ command
+  where
+    values :: Map String AliasArgument
+    values
+      | length arguments == length actualArguments
+        = Map.fromList $ zip arguments actualArguments
+      | otherwise = Map.empty
+
+    replaceArguments :: String -> String
+    replaceArguments s = maybe s toString $ Map.lookup s values
+
+    toString :: AliasArgument -> String
+    toString = \case
+        SimpleArgument str -> str
+        QuotedArgument str -> "\"" <> str <> "\""
+
+createOnePathClaim
+    :: Claim claim
+    => (claim, TermLike Variable)
+    -> Rule.OnePathRule Variable
+createOnePathClaim (claim, cpattern) =
+    Rule.OnePathRule
+    $ Rule.RulePattern
+        { left = cpattern
+        , right = Rule.right . coerce $ claim
+        , requires = Predicate.makeTruePredicate
+        , ensures = Rule.ensures . coerce $ claim
+        , attributes = Default.def
+        }
+
+conjOfOnePathClaims
+    :: [Rule.OnePathRule Variable]
+    -> Sort
+    -> TermLike Variable
+conjOfOnePathClaims claims sort =
+    foldr
+        TermLike.mkAnd
+        (TermLike.mkTop sort)
+        $ fmap Rule.onePathRuleToPattern claims
+
+generateInProgressOPClaims
+    :: Claim claim
+    => MonadState (ReplState claim) m
+    => m [Rule.OnePathRule Variable]
+generateInProgressOPClaims = do
+    graphs <- Lens.use lensGraphs
+    claims <- Lens.use lensClaims
+    let started = startedOPClaims graphs claims
+        notStarted = notStartedOPClaims graphs claims
+    return $ started <> notStarted
+  where
+    startedOPClaims
+        :: Claim claim
+        => Map.Map ClaimIndex ExecutionGraph
+        -> [claim]
+        -> [Rule.OnePathRule Variable]
+    startedOPClaims graphs claims =
+        fmap createOnePathClaim
+        $ claimsWithPatterns graphs claims
+        >>= uncurry zip
+    notStartedOPClaims
+        :: Claim claim
+        => Map.Map ClaimIndex ExecutionGraph
+        -> [claim]
+        -> [Rule.OnePathRule Variable]
+    notStartedOPClaims gphs cls =
+        Rule.OnePathRule
+        . coerce
+        . (cls !!)
+        . unClaimIndex
+        <$> (Set.toList $ Set.difference
+                (Set.fromList $ fmap ClaimIndex [0..length cls - 1])
+                (Set.fromList $ Map.keys gphs)
+            )
+
+claimsWithPatterns
+    :: Map ClaimIndex ExecutionGraph
+    -> [claim]
+    -> [([claim], [TermLike Variable])]
```

Comment:
Is the reason we return an array of `claim` (which always contain a single element) here to make the `uncurry zip` work a few lines up? If so, it's a bit confusing and I think the same result could be achieved with:
```
λ :t (=<<) sequence
(=<<) sequence
  :: (Monad m, Traversable t) => m (t (m a)) -> m (t a)

λ sequence =<< [("a1", ["b", "c"]), ("a2", ["d", "e", "f"])]
[("a1","b"),("a1","c"),("a2","d"),("a2","e"),("a2","f")]
```

So, if you change this to return `[(claim, [TermLike Variable)]`, then you should be able to replace `uncurry zip` with `sequence`.